src/main/java/com/github/tomakehurst/wiremock/matching/RequestPattern.java

```diff
@@ -56,14 +57,41 @@
     private final RequestMatcher defaultMatcher = new RequestMatcher() {
         @Override
         public MatchResult match(Request request) {
-            return MatchResult.aggregate(
-                url.match(request.getUrl()),
-                method.match(request.getMethod()),
-                allHeadersMatchResult(request),
-                allQueryParamsMatch(request),
-                allCookiesMatch(request),
-                allBodyPatternsMatch(request)
+            MatchResult urlMatch = url.match(request.getUrl());
+            MatchResult methodMatch = method.match(request.getMethod());
+            MatchResult headersMatch = allHeadersMatchResult(request);
+            MatchResult queryParamsMatch = allQueryParamsMatch(request);
+            MatchResult cookiesMatch = allCookiesMatch(request);
+            MatchResult bodyPatternsMatch = allBodyPatternsMatch(request);
+            MatchResult overallMatch = MatchResult.aggregate(
+                    urlMatch,
+                    methodMatch,
+                    headersMatch,
+                    queryParamsMatch,
+                    cookiesMatch,
+                    bodyPatternsMatch
             );
+            if (!overallMatch.isExactMatch()) {
+                notifier().info(format("Request does not match pattern:\n" +
+                                "%s\n" +
+                                "Match details: url=%s, method=%s, headers=%s, queryParams=%s, cookies=%s, body=%s",
```

Comment:
I wonder if it's worth going a bit further and printing the expected and actual values where there's a mismatch?